### DISCUSSION TITLE
Is it possible to have conditional code blocks included or excluded at build time?

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [nuxt/nuxt#28818] [Questions]
Hello,
So i know you can have conditional code in the template using v-if and v-show. But all the code still get included at build time, even it not shown.
What I am wanting is an if statement which will be checked at build time ( checking a an environment variable ), and if true a certain code block or template will get included in the final build. If false, it wont get included at all.
Anyone know how to go about this?

**Comment by pixelpaulaus** [comment]
if anyone else is needing this, i found the solution with a VITE plugin:
`vite-plugin-conditional-compiler`

**Comment by xjccc** [comment]
I think it's something like [dev-only components](https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/components/dev-only.ts)
